### PR TITLE
[WIP] Tables

### PR DESCRIPTION
**Pull request by PiotrKozlowski**
Two issues:
- I'm losing all attributes from elements in either serialization or deserialization.
- How to make table cells editable but do not allow to delete them?

Styling for tables: https://github.com/katalysteducation/adaptarr-front/pull/19

### CODE REVIEW

**Comment by aiwenar** [comment]
Could you rebase this on top of master? GitHub seems confused about what's part of this PR and what's not. And in the future, please don't base feature branches on other feature branches.

**Comment by PiotrKozlowski** [comment]
@aiwenar Done

File: src/plugins/table/schema.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+export default {
+    blocks: {
+        table: {
+            data: {
+                summary: s => s == null || typeof s === 'string',
+                frame: f => f == null || typeof f === 'string',
+                colsep: c => c == null || c >= 0,
+                rowsep: r => r == null || r >= 0,
+                pgwide: p => p == null || p >= 0,
+            },
```
Comment (aiwenar) [inline src/plugins/table/schema.js]: Even though it's a temporary solution we might still want a proper schema for those attributes.

File: src/plugins/table/schema.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            nodes: [
+                { match: { type: 'caption' }, min: 0, max: 1, },
+                { match: { type: 'tgroup' }, min: 1, },
+                { match: { type: 'caption' }, min: 0, max: 1, },
+            ],
+            normalize: normalizeTable,
+        },
+        tgroup: {
+            data: {
+                cols: c => c == null || c > 0,
```
Comment (aiwenar) [inline src/plugins/table/schema.js]: `cols` is always required.

**Comment by aiwenar** [comment]
I've also just realised, that rather than adding this temporarily to `cnx-designer`, we could keep it as a plugin in `adaptarr-front`. This would require some changes to CNXML de/serialization to allow for defining custom rules in plugins though.

**Comment by PiotrKozlowski** [comment]
I've opened separate. Now it's handled as separate plugin on the front end - https://github.com/katalysteducation/cnx-designer/pull/19

**Pull request closed without merging**